components/server/src/user/user-controller.ts

```diff
@@ -621,7 +620,7 @@ export class UserController {
             return;
         }
 
-        if (this.urlStartsWith(returnToURL, this.config.hostUrl.toString()) || this.urlStartsWith(returnToURL, this.config.brandingConfig.homepage)) {
```

Comment:
Hm, I'm pretty sure this will break some flows. E.g., login from the website I think.

Reply:
We could move this field up into `config`, and make it optional, and only set it for gitpod.io, for instance.

Reply:
it would also break self-hosted. Very few folks will want to redirect to www.gitpod.io

Reply:
@csweichel Would moving this field up in the config be an incompatible SH change?

Reply:
> Would moving this field up in the config be an incompatible SH change?

No, this should work! :+1:

Reply:
Currently, the installer does not allow to change the `homepage` config value. I don't know in which situation the `redirect` or `returnTo` query variable could point to a page of the homepage. If it's really a use case (@geropl ?), you need to add a `homepage` config value to the [installer config](https://github.com/gitpod-io/gitpod/blob/d9955ccc742430465b62b373baf4a01d3ba11eaf/installer/pkg/config/v1/config.go#L63) as well.

Reply:
The only usage of the `homepage` value seems to check if the redirect to such a page is allowed. For self-hosted, the usage seems to be pretty limited (and not very obvious). If we really need this for SaaS, here are some additional options:

a) always allow redirects to `https://www.gitpod.io/` (hard-coded), just for our SaaS—not very smart but would probably be sufficient
b) allow always `www.$DOMAIN` (just because _we_ need it)—not very smart but would probably be sufficient
c) use a more precise name for the config value, e.g. like `allowAdditionalRedirectTarget` or similar—only if we really need it for self-hosted, but honestly, I don't think it's really a common case worth polluting the installer config for it


But at first, it would make sense to make sure if we really need this for SaaS, in my opinion.

Reply:
> I don't know in which situation the redirect or returnTo query variable could point to a page of the homepage.

Hm, I thought we'd send users to www.gitpod.io on logout, but apprently we don't do that anymore.
IIRC, the "Login" button on www.gitpod.io would rely on the retunrnTo a few months ago (to send users back to some explainer), but that does not seem to be the case anymore. (maybe for 1st-time users?)

We configure(d) branding for production [here](https://github.com/gitpod-io/ops/blob/684defd9d6c0619dd3ab952b5cced3c50f0ad4ec/charts/gitpod_io/values.production.yaml#L131-L132), but:
 - the only value used is `homepage`, making this the only relevant line change in this PR
 - `redirectUrlAfterLogout` (change above in this file) is not used, and `redirectUrlIfNotAuthenticated` not used as well (subject to config drift it seems)
 
 Tl;dr:
  - Logout: we don't seem to send users to external websites anymore (stays on "Login" screen)
  - we don't seem to need it for website-flows (maybe @jldec can rule this out for first-time users?)
  - this change should be fine, then :shrug:

Reply:
In order not to break SaaS, I've hard coded `https://www.gitpod.io` as an "allowed-return-to-URL-prefix" (option a in @corneliusludmann's proposal). I think this is not worthwhile to be part of (SH) configuration, unless we change the redirect logic more deeply.